### PR TITLE
Add per-event checklist to event settings

### LINKED ISSUE #121: [IDEA] Add Co-Host "Checklist"
Idea from Kim White and Mervin -- we wish we had a mutually visible checklist between sponsors and organizers, in order to complete all of the steps required to participate in Camp. Inquiry request, review of their OS compliance, accepting/offering sponsorship spot, receipt of payment information, issuing invoice, paying invoice, publishing their participation, etc.

### PR DESCRIPTION
**Pull request by faisalahammad**
Fixes #121

## Proposed changes:

* Adds a new `gatherpress-event-checklist` post type support, declared on `gatherpress_event` next to the existing event supports.
* Adds `Event\Checklist`, which registers the `gatherpress_checklist` post meta as a JSON string of ordered `{ id, text, completed }` items. The meta is registered with `show_in_rest` in the `edit` context only, so payment and compliance notes never leave the editor through public REST reads. `sanitize_callback` runs on every write: it drops non-array payloads to `[]`, drops items without a usable id, sanitizes text with `sanitize_text_field`, coerces `completed` through `rest_sanitize_boolean`, and caps the list at 200 items and 255 characters per item.
* Adds a Checklist section to the event settings panel (`src/panels/event-settings/checklist/`), gated with `usePostTypeSupports` so it only renders on post types that declare the support. It has a text field and completed checkbox per row, add and remove, move up and down, and an n-of-m progress label. All string handling sits in `helpers.js` as pure functions so the logic is unit-testable.
* Documents the new support in `AGENTS.md` and `docs/developer/post-type-supports/README.md`.

Not in this pass: the issue also asks for the checklist to be visible to sponsors. That needs a participant and permission model that does not exist yet, so it is left out here. The stored item shape has room for assignment and due date fields later without a migration.

### Other information:

- [x] Have you written new tests for your changes, if applicable?

Tests: `test/unit/php/includes/tests/core/classes/event/class-test-checklist.php` (16 tests, full statement and method coverage of the new class) and `test/unit/js/src/panels/event-settings/checklist/helpers.test.js` (44 tests, full helper coverage).

## Testing instructions:

* Go to Events and edit an event, or create a new one.
* Open the Event Settings panel and find the Checklist section.
* Add a few items matching the checklist on the issue, for example "Inquiry received", "OS compliance review", "Sponsorship accepted", "Payment details", "Invoice issued", "Invoice paid", "Participation published".
* Tick some items done and watch the progress label change.
* Reorder with the up and down buttons, and delete one with the remove button. The first item's up button and the last item's down button should be disabled.
* Save the event and reload. The items, order, and completed state should come back the same.
* View the event on the front end while logged out. Nothing about the checklist should show, and it should not appear in the REST response for the event either.
* Log in as another editor, open the same event, and confirm you see the same checklist.

## Before / After

| Before | After |
| ------ | ----- |
| The Event settings panel had no checklist section. There was no place in the editor to track the steps for running an event. | The Event settings panel has a Checklist section: one row per item with a completed checkbox and a text field, plus add, remove, move up and move down buttons. A progress line above the list reads `n of m complete`, or shows a short prompt while the list is empty. |

The change is visible only in the block editor. Nothing about the checklist renders on the front end, and the stored value stays out of public REST reads.

### Credit (for release notes)

My WordPress.org username: faisalahammad

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Add an event checklist to the event settings panel, stored in post meta and readable only in the REST edit context.

</details>

The changelog entry file is committed on this branch at `.github/changelog/121-event-checklist`, since the workflow job cannot push to a fork. I left the auto-generate box unchecked so the two entries do not collide.

## Summary by CodeRabbit

* **New Features**
  * Added an organizer checklist to event settings.
  * Organizers can add, edit, complete, reorder, and remove checklist items.
  * Checklist progress is displayed, with limits of 200 items and 255 characters per item.
  * Checklist data is available through editing workflows while remaining excluded from public event views.

* **Documentation**
  * Documented checklist support, usage, storage, and editing behavior.

* **Tests**
  * Added coverage for checklist functionality, validation, limits, and edge cases.

### CODE REVIEW

File: includes/core/classes/event/class-checklist.php [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	 * @return string JSON-encoded checklist.
+	 */
+	public function sanitize( $value ): string {
+		if ( ! is_string( $value ) ) {
+			return self::EMPTY_CHECKLIST;
+		}
+
+		$decoded = json_decode( $value, true );
+
+		if ( ! is_array( $decoded ) ) {
```
[the remark this answers is not in the document]
Comment (faisalahammad) [reply includes/core/classes/event/class-checklist.php]: Fixed in b977e9c5. sanitize() now also checks array_is_list(), so an object payload falls back to the empty list instead of being reindexed. Covered by test_sanitize_rejects_associative_payload.

File: includes/core/classes/event/class-checklist.php [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	 * @param mixed $item Raw item as submitted.
+	 *
+	 * @return array{id: string, text: string, completed: bool}|null Sanitized item, or null when unusable.
+	 */
+	protected function sanitize_item( $item ): ?array {
+		if ( ! is_array( $item ) || ! isset( $item['id'] ) || ! is_scalar( $item['id'] ) ) {
+			return null;
+		}
+
+		$id = sanitize_text_field( (string) $item['id'] );
```
[the remark this answers is not in the document]
Comment (faisalahammad) [reply includes/core/classes/event/class-checklist.php:262]: Fixed in b977e9c5. Added Checklist::MAX_ID_LENGTH = 64 and sanitize_item() drops items with longer ids. The editor helpers carry the same cap (MAX_CHECKLIST_ID_LENGTH) so both sides stay in sync. Tests cover over the cap and at the cap.

File: src/panels/event-settings/checklist/index.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+							/>
+							<Button
+								icon={ chevronDown }
+								label={ __( 'Move down', 'gatherpress' ) }
+								disabled={ index === total - 1 }
+								onClick={ () =>
+									commit( moveChecklistItem( items, item.id, 1 ) )
+								}
+							/>
+							<Button
```
[the remark this answers is not in the document]
Comment (faisalahammad) [reply src/panels/event-settings/checklist/index.js:158]: Added in b977e9c5. test/unit/js/src/panels/event-settings/checklist/index.test.js covers the panel: support gate, empty state, progress line, text edit, checkbox toggle, add, remove, move up, move down, and disabled boundary buttons. Also added event-settings/index.test.js to keep the panel mounted inside the Event settings integration.

File: test/unit/php/includes/tests/core/classes/event/class-test-checklist.php [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+	 */
+	public function test_sanitize_falls_back_when_text_is_not_scalar(): void {
+		$instance = Checklist::get_instance();
+
+		$result = $instance->sanitize( '[{"id":"a","text":{"nested":true},"completed":false}]' );
+
+		$this->assertSame(
+			'[{"id":"a","text":"","completed":false}]',
+			$result,
+			'Failed to assert a non-scalar text is discarded.'
```
[the remark this answers is not in the document]
Comment (faisalahammad) [reply test/unit/php/includes/tests/core/classes/event/class-test-checklist.php:330]: Added in b977e9c5. test_sanitize_coerces_non_scalar_completed_to_false sends an object and an array as completed and checks both come out as false.

**Review by faisalahammad — COMMENTED** [review]
All review items are addressed in b977e9c5.

- sanitize() rejects non-list JSON payloads with array_is_list() so an object no longer gets reindexed as a list.
- Item ids are capped at 64 chars via Checklist::MAX_ID_LENGTH, mirrored in the editor helpers.
- Added panel tests and an Event settings integration test for the checklist UI.
- Added coverage for non-scalar completed values.
- Added REST tests for the view/edit context and write permissions, which the review body flagged as a nitpick.

PHP: 24 tests pass with 100 percent coverage on class-checklist.php. JS: 61 tests pass. PHPCS, PHPStan, ESLint, and CodeRabbit are clean.

**Comment by faisalahammad** [comment]
Pushed fixes for the failing checks. Two commits: one for the REST privacy bug, one for the two CI jobs that broke repo-wide.

**The REST failure**

`Test_Checklist::test_rest_view_context_omits_the_checklist` failed on all four PHP versions with the checklist key still in a public `view` response.

Cause: `register_post_meta()` gates writes through `auth_callback`, but a registered value goes to anyone who can read the post. Core's context filter drops keys it finds a schema property for and keeps the ones it does not, so `context => edit` was the whole privacy promise and it depended on the events controller's item schema being rebuilt after this meta was registered. That schema is memoized per controller, so a controller built earlier carries an empty meta schema and hands the checklist to a public read. This is why it only showed up in the full suite and not in a filtered run.

Fix: `Checklist::strip_from_readers()` on `rest_prepare_<post type>` unsets the checklist when `current_user_can( 'edit_post', $post->ID )` is false. It runs after core's context filter, so the answer comes from the capability and not from schema internals. Same pattern `Event\Rest_Api::prepare_event_data()` already uses for the online event link. The existing view test passes unchanged, and the edit and permission tests cover the keep branch.

| Check | Cause | Fix |
| --- | --- | --- |
| PHPUnit x4 | Checklist in public REST response | `strip_from_readers()` capability check |
| Coverage | Failed run left `coverage.xml` missing | Clears with the PHPUnit fix |
| wp.org guidelines | `dist-archive-command:^2` branch deleted upstream | `composer config --global use-github-api false` |
| Spell Check with Typos | `crate-ci/typos@master` branch renamed to `main` | Pinned to `v1.50.2` |

The last two broke on their own between 12:01 and 16:18 UTC today, not from anything in this branch. I checked the run history to confirm. The typos action no longer resolves at all, and the wp.org job now falls back to an SSH clone because composer.json 404s on the missing branch. Both fixes are in `.github/workflows/` only.

**Local verification**

PHPCS clean, PHPStan clean, full PHPUnit 2113 tests pass, multisite suite passes, coverage gate green with `class-checklist.php` at 100%. The `Test_Event_Email` failure that appears in the old log after coverage generation is wrapper noise, it passes locally in both filtered and full runs.

I also corrected the REST path in `TESTING_INSTRUCTIONS.md`, it had `gatherpress_event` where the base is `gatherpress_events`.
